webknossos-datastore/app/com/scalableminds/webknossos/datastore/services/BinaryDataService.scala

```diff
@@ -108,16 +108,52 @@ class BinaryDataService(val dataBaseDir: Path,
                                                   conversionFunc(inputArray))
     else Fox.successful(inputArray)
 
+  /*
+   * Everything outside of the layer bounding box is set to black (zero) so data outside of the specified
+   *  bounding box is not exposed to the user
+   */
+  private def clipToLayerBoundingBox(request: DataServiceDataRequest)(inputArray: Array[Byte]): Box[Array[Byte]] = {
+    val bytesPerElement = request.dataLayer.bytesPerElement
+    val requestBboxInMag = request.cuboid.toBoundingBoxInMag
+    val layerBboxInMag = request.dataLayer.boundingBox / request.mag // Note that this div is implemented to round to the bigger bbox so we don’t lose voxels inside.
+    val intersectionOpt = requestBboxInMag.intersection(layerBboxInMag).map(_.move(-requestBboxInMag.topLeft))
+    val outputArray = Array.fill[Byte](inputArray.length)(0)
+    intersectionOpt.foreach { intersection =>
+      for {
+        z <- intersection.topLeft.z until intersection.bottomRight.z
+        y <- intersection.topLeft.y until intersection.bottomRight.y
+        // We can bulk copy a row of voxels and do not need to iterate in the x dimension
+      } {
+        val offset =
+          (intersection.topLeft.x +
+            y * requestBboxInMag.width +
+            z * requestBboxInMag.width * requestBboxInMag.height) * bytesPerElement
+        System.arraycopy(inputArray,
+                         offset,
+                         outputArray,
+                         offset,
+                         (intersection.bottomRight.x - intersection.topLeft.x) * bytesPerElement)
```

Comment:
IMO this is the most likely cause for the nullpointer

Reply:
some out of bounds maybe. Maybe even due to my comment above regarding them dimensions not being converted to the respective mag

Reply:
For the record, the `null` was explicitly set for request.dataSource, compare https://github.com/scalableminds/webknossos/pull/8573#issuecomment-2838145101